### ISSUE TITLE
 java.lang.IllegalArgumentException: Illegal URL  at retrofit2.Retrofit$Builder.baseUrl(Retrofit.java:437)

### ISSUE THREAD

**Issue by riyal123** [square/retrofit#1872]
Getting this error on pre Lollypop devices. 

I tested this on HTC Desire 500 Android 4.1

Works fine for post 5.0 devices

**Comment by swankjesse** [comment]
What's the URL?

**Comment by riyal123** [comment]
its http://www.mocky.io/v2/

same works for android 5.0

**Comment by swankjesse** [comment]
That's surprising. Could you provide a failing test case?

**Comment by riyal123** [comment]
public static String domainToAscii(String input) {
    try {
      **String result = IDN.toASCII(input).toLowerCase(Locale.US);**
      if (result.isEmpty()) return null;

      // Confirm that the IDN ToASCII result doesn't contain any illegal characters.
      if (containsInvalidHostnameAsciiCodes(result)) {
        return null;
      }
      // TODO: implement all label limits.
      return result;
    } catch (IllegalArgumentException e) {
      return null;
    }
  }

This line gives illegal argument exception U_FILE_ACCESS_ERROR

**Comment by kenumir** [comment]
I received the same error from production app from device: samsung/GT-I9300/android 4.3

```
Caused By: java.lang.IllegalArgumentException·Illegal URL: http://api.<my api domain>.io/v1/
```

I try testing on similar devices but no error occurs.
Any progress with that?

**Comment by swankjesse** [comment]
@kenumir can you provide a failing test case?

**Comment by kenumir** [comment]
@swankjesse Unfortunately no, error comes to bugsnag, for now only once.

**Comment by swankjesse** [comment]
Please open a new issue when you can provide a failing test case.

**Comment by swankjesse** [comment]
Forgot the `http://` prefix?

**Comment by prachi0311** [comment]
No I'm using the same baseurl in postman...when I try it with http:// I'm getting 410 response.

**Comment by satheesh93** [comment]
@prachi0311 have you fixed this error? If so, Please share.

**Comment by saqibmirza2007** [comment]
I got same issue as I have added an letter in port and the app got crashed
Process: com.signinghub.nd, PID: 14379
    java.lang.IllegalArgumentException: Invalid URL port: "82i"
        at okhttp3.HttpUrl$Builder.parse$okhttp(HttpUrl.kt:1330)
        at okhttp3.HttpUrl$Companion.get(HttpUrl.kt:1634)
        at okhttp3.HttpUrl.get(Unknown Source:2)
        at retrofit2.Retrofit$Builder.baseUrl(Retrofit.java:492)
        at com.signinghub.sdk.apiclients.RetroAPIClient.getGeneralClient(RetroAPIClient.kt:24)
        at com.signinghub.datamodels.PublicProfilesDM.observePublicProfilesCall(PublicProfilesDM.kt:30)
        at com.signinghub.datamodels.PublicProfilesDM.getPublicProfiles(PublicProfilesDM.kt:21)
        at com.signinghub.viewmodels.PublicProfilesVM.getPublicProfiles(PublicProfilesVM.kt:15)
        at com.signinghub.views.fragments.LoginFragment.onViewCreated(LoginFragment.kt:47)
        at androidx.fragment.app.FragmentStateManager.createView(FragmentStateManager.java:332)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1187)
        at androidx.fragment.app.FragmentManager.addAddedFragments(FragmentManager.java:2224)
        at androidx.fragment.app.FragmentManager.executeOpsTogether(FragmentManager.java:1997)
        at androidx.fragment.app.FragmentManager.removeRedundantOperationsAndExecute(FragmentManager.java:1935)
        at androidx.fragment.app.FragmentManager.execPendingActions(FragmentManager.java:1849)
        at androidx.fragment.app.FragmentManager$4.run(FragmentManager.java:413)
        at android.os.Handler.handleCallback(Handler.java:873)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:214)
        at android.app.ActivityThread.main(ActivityThread.java:7099)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)

**Comment by rickyManalo** [comment]
So, I'm trying use Retrofit to call that free NASA API, but it throws this error. The url looks like this: https://api.nasa.gov/planetary/apod?date=2023-02-10&api_key=<api key> I created it using the Uri.Builder(). As you can see, the apod part goes straight to a question mark, but it works fine when you use it on a browser. If I manually create the url with an extra / after apod, I think the url works.
